http://codereview.qt-project.org/#change,16809
Change I469e3d39: use a translucent drag pixmap if compositing is active
by Ignat Semenov
Upstreamed: due to be accepted after Qt 5 d'n'd is fixed(!)

---
 src/gui/kernel/qdnd_x11.cpp |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

--- a/src/gui/kernel/qdnd_x11.cpp
+++ b/src/gui/kernel/qdnd_x11.cpp
@@ -285,11 +285,15 @@ public:
 
     void setPixmap(const QPixmap &pm)
     {
-        QBitmap mask = pm.mask();
-        if (!mask.isNull()) {
-            setMask(mask);
-        } else {
+        if (QX11Info::isCompositingManagerRunning()) {
+            setAttribute(Qt::WA_TranslucentBackground);
             clearMask();
+        } else {
+            QBitmap mask = pm.mask();
+            if (!mask.isNull())
+                setMask(mask);
+            else
+                clearMask();
         }
         resize(pm.width(),pm.height());
         pixmap = pm;
